src/main/java/com/clouway/cuse/spi/SearchEngineImpl.java

```diff
@@ -56,4 +68,14 @@ public void delete(String indexName, List<Long> objectIds) {
     indexRegistry.delete(indexName, objectIds);
   }
 
+  private boolean isWithEquivalentSearchIndexes(List<? extends Object> objects) {
+    Annotation searchIndexToMatch = objects.get(0).getClass().getAnnotations()[0];
```

Comment:
SearchIndex annottation  = clazz.getAnnotation(SearchIndex.class);
String indexName = annottation.name();

Then check the other if their index names match with the first one